### PR TITLE
Add page and application structure doc

### PR DESCRIPTION
**Pull request by jbalboni**
I'm not entirely sure this is exactly what we want out of this ticket, but it seems like useful information that is sort of related to application structure and boundaries.

Screen Shot 2019-05-22 at 3 54 58 PM

### CODE REVIEW

File: packages/documentation/src/pages/platform/site-structure/application-structure.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
+---
+title: Page and application Javascript structure
+tags: page, bundle, react, javascript, structure, webpack, site-wide
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/site-structure/application-structure.mdx]: you can leave out words that are in the title unless you want them to show up as matching tags

Screen Shot 2019-05-23 at 08 50 42

File: packages/documentation/src/pages/platform/site-structure/application-structure.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+## Structure of a page
+
+A typical application page has several independent React components. Here's an abbreviated screenshot of the `/my-va` page:
+
+![Abbreviated dashboard with red boxes drawn around React mount points](../../../images/platform/react-mount-points.png)
+
+Each of the areas marked with a red box is a separately mounted React component. These React components shared a common Redux store:
+
+```mermaid
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/site-structure/application-structure.mdx]: this didn't render for me locally

File: packages/documentation/src/pages/platform/site-structure/application-structure.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+![Abbreviated dashboard with yellow boxes drawn around site wide components](../../../images/platform/sitewide_primary.png)
+
+The components in the yellow boxes are the common site-wide components started by [startSitewideComponents](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/site-wide/index.js#L26) and the blue box is the primary React application.
+
+## Structure of a React application
+
+Our React applications can vary in structure depending on what their purpose is, but typically they have a reducer and actionsfor Redux related logic and routes and components for the rest of the application. Routes are typically the best place to start looking at an application. Again using `/my-va` as an example, you can see the React components to start looking at in the [routes.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/personalization/dashboard/routes.jsx) file:
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/site-structure/application-structure.mdx]: `actions for`

File: packages/documentation/src/pages/platform/site-structure/application-structure.mdx (hunk 2 above)
Comment (jbalboni) [reply packages/documentation/src/pages/platform/site-structure/application-structure.mdx]: It seems to break if there's a change to the page, but it renders for me whenever I restart the `develop` process:

Screen Shot 2019-05-23 at 12 04 57 PM

**Pull request merged**